src/utils/integration/metadata.ts

```diff
@@ -0,0 +1,43 @@
+import { exists, fs } from "../../fs.js";
+
+interface PrismMetadataOptions {
+  fromDist?: boolean;
+}
+
+const CNI_METADATA_RELATIVE_PATH = ".spectral/prism.json";
+
+function getPrefix(fromDist = false) {
+  return fromDist ? "../" : "./";
+}
+
+export async function getPrismMetadata(
+  options: PrismMetadataOptions = {},
+): Promise<Record<string, string>> {
+  const metadataPath = `${getPrefix(options.fromDist)}${CNI_METADATA_RELATIVE_PATH}`;
+  let metadata = {};
+
+  if (await exists(metadataPath)) {
+    const parsed = JSON.parse(await fs.readFile(metadataPath, { encoding: "utf-8" }));
+    metadata = parsed;
+  }
+
+  return metadata;
+}
```

Comment:
```suggestion
export async function getPrismMetadata(
  options: PrismMetadataOptions = {},
): Promise<Record<string, string>> {
  const metadataPath = `${getPrefix(options.fromDist)}${CNI_METADATA_RELATIVE_PATH}`;

  if (!(await exists(metadataPath))) {
    return {};
  }

  try {
    const content = await fs.readFile(metadataPath, { encoding: "utf-8" });
    return JSON.parse(content);
  } catch (error) {
    console.warn(`[Warning] Failed to parse metadata at ${metadataPath}:`, error);
    return {};
  }
}
```

Lets add a try/catch around json.parsee, remove the mutable binding, and do early returns.